snc.sh

```diff
@@ -35,6 +36,64 @@ else
     fi
 fi
 
+function preflight_failure() {
+        local msg=$1
+        echo "$msg"
+        if [ -z "${SNC_NON_FATAL_PREFLIGHT_CHECKS-}" ]; then
+                exit 1
+        fi
+}
+
+function run_preflight_checks() {
+        echo "Checking libvirt and DNS configuration"
+
+        LIBVIRT_URI=qemu+tcp://localhost/system
+
+        # check if libvirtd is listening on a TCP socket
+        if ! virsh -c ${LIBVIRT_URI} uri >/dev/null; then
+                preflight_failure  "libvirtd is not listening for plain-text TCP connections, see https://github.com/openshift/installer/tree/master/docs/dev/libvirt#configure-libvirt-to-accept-tcp-connections"
+        fi
+
+        # check if libvirtd has access to an x86_64 hypervisor
+        if ! virsh -c ${LIBVIRT_URI} capabilities | ${XMLSTARLET} sel -t -c '/capabilities/guest/arch[@name="x86_64"]' >/dev/null; then
```

Comment:
Shouldn't this check will fail for the folks who want to use snc for powerpc or other arch ?

Reply:
Yes it will fail for these users, which is why there's a way to disable the failed preflights.
I'd need to change this test to
```
arch=$(virsh  capabilities | xmlstarlet sel -t -c '/capabilities/host/cpu/arch')
if [ arch != "x86_64 ]; then
    # warn
fi
```
and then use $arch in the kvm test instead of hardcoding `x86_64`